app/lib/app-extension/Extension.js

```diff
@@ -301,11 +302,18 @@ module.exports = class Extension {
     )
   }
 
+  __getPkgSrc () {
```

Comment:
The main goal is to respect the entrypoint in `package.json`. Right now it is hardcoded to `src`. Now I think of it this solution also isn't ideal, as it won't be possible to use a nested folder for `dist`. Assuming `dist/index.js` is the compiled version of `src/index.ts`, it will work fine.

Ideally it would look up the entrypoints from the `exports` field in `package.json`, but this isn't supported in the starter kits at the moment.